spring-core/src/main/java/org/springframework/util/FastByteArrayOutputStream.java

```diff
@@ -162,9 +163,26 @@ public void close() {
 	 */
 	@Override
 	public String toString() {
-		return new String(toByteArrayUnsafe());
+		return toString(Charset.defaultCharset());
 	}
 
+	/**
+	 * Converts the buffer's contents into a string by decoding the bytes using
+	 * the specified {@link java.nio.charset.Charset charset}.
+	 *
+	 * @param      charset  the {@linkplain java.nio.charset.Charset charset}
+	 *             to be used to decode the {@code bytes}
+	 * @return     a String decoded from the buffer's contents
+	 */
+	public String toString(Charset charset) {
```

Comment:
For future reference, please keep in mind that we do not indent Javadoc tag content and that we need to add a `@since` tag for new elements in the public API.

However, I'll address those issues when merging, so there's no need to update this PR.

Reply:
In addition, references to instance fields must be prefixed with `this.`.

See Spring Framework's [Code Style](https://github.com/spring-projects/spring-framework/wiki/Code-Style) documentation for details.